scripts/tank_cmd.py

```diff
@@ -189,19 +197,36 @@ def show_help(log):
         log.info(x)
 
 
-def ensure_authenticated():
+def ensure_authenticated(cmd_line_credentials):
     """
     Make sure that there is a current toolkit user set.
     May prompt for a login/password if needed
+
+    :param cmd_line_credentials: Dictionary of credentials passed from the
+                                 command line. May be empty or None.
     """
     # create a core-level defaults manager.
     # this will read site details from shotgun.yml
     core_dm = CoreDefaultsManager()
     # set up the authenticator
     shotgun_auth = ShotgunAuthenticator(core_dm)
-    # request a user, either by prompting the user or by pulling out of
-    # saves sessions etc.
-    user = shotgun_auth.get_user()
+
+    if cmd_line_credentials:
```

Comment:
I am missing a comment somewhere in the code which explains to readers that if `cmd_line_credentials` are provided, these will be used directly - but not remembered by the authentication system. But the default mode, where the authentication system prompts for your username, is trying to remember the details for the next time round!